sdk/core/azure-core/src/http/curl/curl.cpp

```diff
@@ -38,11 +38,18 @@ inline void LogThis(std::string const& msg)
 }
 
 template <typename T>
+#if defined(_MSC_VER)
+#pragma warning(push)
+#pragma warning(disable : 26812)
```

Comment:
Add a comment for what warning 26812 is.  Brief description and link to warning documentation for msvc.  Helps for future understanding when reading code.

Reply:
Why are we suppressing these warnings in the source code? I thought the issue was specific to tests. Should we fix the source instead?

Reply:
I just went further and fixed this warning for our source as well. There were only two instances. The warning says, `CURLcode` (defined by CURL library )is a C-style `enum`. Use `enum class`. We don't have control over that, so we fix in the code.

Reply:
Makes sense, thanks!